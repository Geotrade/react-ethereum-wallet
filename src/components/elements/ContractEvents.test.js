import React from 'react';
import renderer from 'react-test-renderer';

import fetchMock from 'fetch-mock';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('contract events', () => {
  afterEach(() => {
    fetchMock.restore();
  });

  it('renders date td', () => {
    const tree = renderer
      .create(
        <td
          className="time simptip-position-right simptip-movable"
          data-tooltip="//TODO: get timestamp"
        >
          <h2>Nov</h2>
          <p>16</p>
        </td>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // TODO:
  // it('renders log event', () => {
  //   const tree = renderer
  //     .create(
  //       <td className="account-name">
  //         <h2>{props.log.event}</h2>
  //         <p
  //           key="1"
  //           style={{ wordBreak: 'break-word' }}
  //         >
  //           {props.val} : &nbsp; <strong> {props.log.returnValues[props.val]}</strong>
  //           <br />
  //         </p>
  //       </td>
  //     )
  //     .toJSON();
  //   expect(tree).toMatchSnapshot();
  // });

  // TODO
  // it('renders emptylog event', () => {
  //   const tree = renderer
  //     .create(
  //       <td className="account-name">
  //         <h2>{props.log.event}</h2>
  //       </td>
  //     )
  //     .toJSON();
  //   expect(tree).toMatchSnapshot();
  // });

  // TODO
  // it('renders whole component', () => {
  //   const tree = renderer
  //     .create(
  //       <table className="dapp-zebra transactions">
  //       <tbody>
  //         {logs.map(log => (
  //           <tr
  //             key="1"
  //             onClick={{modals: {displayEventModal: true}}}
  //             data-transaction-hash="0x0000000000000000000000000000000000000000"
  //             data-block-hash="0x0000000000000000000000000000000000000000"
  //           >
  //             <DateFormat log={log} />
  //             <Event log={log} />
  //           </tr>
  //         ))}
  //       </tbody>
  //     </table>
  //     )
  //     .toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
});
