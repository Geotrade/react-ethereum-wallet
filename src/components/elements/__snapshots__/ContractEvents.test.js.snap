// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contract events renders date td 1`] = `
<td
  className="time simptip-position-right simptip-movable"
  data-tooltip="//TODO: get timestamp"
>
  <h2>
    Nov
  </h2>
  <p>
    16
  </p>
</td>
`;
